src/DSharp.Mscorlib/Array.cs

```diff
@@ -33,35 +37,48 @@ public sealed class Array : ICollection
 
         public extern Array Filter(ArrayItemFilterCallback itemFilterCallback);
 
+        // I guess these two fro each methods are just js forEach.
+        // ForEach in CLR looks like this -> static void ForEach<T>(T[] array, Action<T> action)
         public extern void ForEach(ArrayCallback callback);
 
         public extern void ForEach(ArrayItemCallback itemCallback);
 
         public extern IEnumerator GetEnumerator();
 
+        // No GetRange methods in CLR
         public extern Array GetRange(int index);
 
         public extern Array GetRange(int index, int count);
 
+        // IndexOf is always static in CLR
+        // The most similar in CLR is -> static int IndexOf(Array array, object? value)
         public extern int IndexOf(object item);
 
+        // The most similar in CLR is -> static int IndexOf(Array array, object? value, int startIndex)
         public extern int IndexOf(object item, int startIndex);
 
+        // These Join methods should be implemented as part of Linq. There's no Join methods on Array in CLR
         public extern string Join();
 
         public extern string Join(string delimiter);
 
+        // LastIndexOf is always static in CLR
+        // The most similar in CLR is -> static int LastIndexOf(Array array, object? value)
         public extern int LastIndexOf(object item);
```

Comment:
Might be worth checking we don't have any compiler magic - Pretty sure we don't, so we can deprecate this!